### PR TITLE
Fixed issue in exporting inventory

### PR DESCRIPTION
**Pull request by tarunbhardwaj**

### CODE REVIEW

File: product.py [hunk 1]
```diff
@@ -262,9 +262,13 @@ def export_bulk_inventory(cls, listings):
         """
         bulk export inventory to amazon
         """
-        channel = (listings and listings[0].channel) or None
+        if not listings:
+            # Nothing to update
+            return
 
-        if channel and channel.source != 'amazon_mws':
+        channel = listings[0].channel
```
Comment (sharoonthomas) [inline product.py]:
```non_amazon_listings = Listing.search[
  ('id', 'in', map(int, listings)),
  ('channel.source', '!=', 'amazon_mws')
]
```

and call super with that

Comment (tarunbhardwaj) [inline product.py]: This method is called from a method inside a channel so here we expect all listing should belongs to same channel.
If not then there can be more than one amazon channel listings and then super call for non amazon channel is fine but further logic of this method is not to handle multiple amazon channel at once.

File: product.py [hunk 2]
```diff
@@ -262,19 +262,34 @@ def export_bulk_inventory(cls, listings):
...
+        ])
+        if non_amazon_listings:
             return super(ProductSaleChannelListing, cls).export_bulk_inventory(
-                listings
+                non_amazon_listings
             )
+        amazon_listings = filter(
+            lambda l: l not in non_amazon_listings, listings
+        )
+        if not amazon_listings:
```
Comment (sharoonthomas) [inline product.py]: not required. the for loop will exit on its own

File: product.py [hunk 3]
```diff
@@ -262,19 +262,34 @@ def export_bulk_inventory(cls, listings):
...
+            # XXX: No listing to export
+            return
 
-        inventory_xml = []
-        for listing in listings:
+        inventory_channel_map = {}
```
Comment (sharoonthomas) [inline product.py]: inventory_channel_map = defaultdict(list)

File: product.py [hunk 4]
```diff
@@ -262,19 +262,34 @@ def export_bulk_inventory(cls, listings):
...
+        for listing in amazon_listings:
             product = listing.product
+            channel = listing.channel
 
+            # group inventory xml by channel
+            inventory_channel_map.setdefault(channel, [])
```
Comment (sharoonthomas) [inline product.py]: remove

**Pull request merged**